### ISSUE TITLE
EPIC - CLI : ps-upgrade:check-compatibility - implement check-compatibility

### ISSUE THREAD

**Issue by fabienvallon** [PrestaShop/PrestaShop#35162]
As a user I want to know the my system is compatibile with the new version of PrestaShop  so I can  upgrade prestashop

## Description

Implements **ps-upgrade:check-compatibility - \<TARGET_VERSION\> --checklist \<CHECKLIST\>**

Check the compatibility for a TARGET version 

With 
\<TARGET_VERSION\> : targeted vesion
\<CHECKLIST\> : list of the checks (mem, cpu, swap, disk, php-version, php-ext, php-settings, db-version) 

### Help

Message

_ps-upgrade:check-compatibility , check the compatibility for a prestashop version
Usage: ps:upgrade:check-compatibility   --checklist [ARGS_LIST]   \<TARGET_VERSION\> 
With \<TARGET_VERSION\> the prestashop version you want to verify
--checklist separate by comma : php-version, php-ext, acces-db 
-verbose: sets the verbosity level (e.g. 1 the default, 2 and 3, or you can use respective shortcuts -v, -vv and -vvv)
--quiet: disables output and interaction
--no-interaction: disables interaction
--version: outputs the version number of the console application
--help: displays the command help
--ansi|--no-ansi: whether to force of disable coloring the output_ 

### Global options 

See [Global options](https://symfony.com/doc/current/console/input.html#command-global-options)

### memory (mem)
Optional : not in MVP

### cpu (cpu)
Optional : not in MVP

### swap (swap)
Optional : not in MVP

### disk space (disk)
Optional : not in MVP

### php-version (php-version)
Check PHP compatibility

### php-ext (php-ext)
Check php-ext version compatibility

### php-settings (php-settings)
Check php-settings (such as  memory_limit)  compatibility

### database version (db-version)
Optional : not in MVP

### database access (db-access)
Check DB access

### Message (STDOUT)

For each test (specify by checklist or all ) one per line 

\<TEST NAME\> .....\<OK|KO\> : \<DESCRIPTION if KO\>

Examples
`php-version ....OK 
`
`php-version ....KO : prestashop-9.0.1 required PHP >= 8.1
`

A full example 
```
$ check-compatibility  --check-list=mem,cpu,php-version,db,db-access /--package=/somewhere/prestashop-9.0.1.zip
mem....SKIP  : Can not be test
cpu: OK
php-version: KO - please upgrade your system to PHP >= 8.1
php-ext : OK
```

  
### Accestance test : help 

```
$ ps-upgrade:check-compatibility --help
ps-upgrade:check-compatibility , check the compatibility for a prestashop version
Usage: ps:upgrade:check-compatibility   --checklist [ARGS_LIST]   \<TARGET_VERSION\> 
With \<TARGET_VERSION\> the prestashop version you want to verify
--checklist separate by comma : php-version, php-ext, acces-db 
-verbose: sets the verbosity level (e.g. 1 the default, 2 and 3, or you can use respective shortcuts -v, -vv and -vvv)
--quiet: disables output and interaction
--no-interaction: disables interaction
--version: outputs the version number of the console application
--help: displays the command help
--ansi|--no-ansi: whether to force of disable coloring the output
```

### Acceptance test : OK 

TODO jeu de tests 

```
$ ps-upgrade:check-compatibility  --check-list=php-version,php-ext,db-access  9.0.0
php-version: OK
php-ext : OK
db-access: OK 
```

 ### Acceptance test : KO 

TODO per test 

```
$ ps-upgrade:check-compatibility  --check-list=php-version,php-ext,db-access 9.0.0
php-version: KO <ERROR>
php-ext : OK <ERROR>
db-access: OK <ERROR>
```

| Check type| Errors |
|----------------|----------|
| php-version | please upgrade your system to PHP >= A.B  and <= C.D | 
| php-version | please downgrade your system to PHP >= A.B  and <= C.D  | 
| php-ext | please upgrade your <EXT> to  >= A.B  and <= C.D | 
| php-ext  * | please downgrade your <EXT> to >= A.B  and <= C.D  | 
| php-ext | please downgrade your <EXT> to >= A.B  and <= C.D  | 
| php-settings * | <SETTINGS> Should be <MESSAGE>  | 
| db-access * | TODO  | 

* You could have multiple error  = multiple extension / settings (multiline) 

### Technical specs

- [x] #35163
- [x] #35164

**Comment by Hlavtox** [comment]
Ping @Quetzacoalt91 Thomas, can I close this?

**Comment by Quetzacoalt91** [comment]
Yep, this has been replaced by Update Assistant 7. On the CLI: `bin/console update:check-requirements [admin]` will do the job
